### PR TITLE
frontend: Remove SetRepositoryEnabled mutation

### PR DESCRIPTION
**Pull request by ryanslade**
It has been deprecated for a while and we have confirmed that it is not being
used anymore.

As a side effect, we no longer need any of the code to programatically exclude
repos from external service config which allows us to remove a lot of dead
code.

Closes https://github.com/sourcegraph/sourcegraph/issues/21013

### CODE REVIEW

File: cmd/frontend/graphqlbackend/repositories.go [hunk 1]
```diff
@@ -272,41 +269,6 @@ func (r *repositoryConnectionResolver) PageInfo(ctx context.Context) (*graphqlut
...
-		return nil, err
-	}
-
-	repo, err := r.repositoryByID(ctx, args.Repository)
-	if err != nil {
-		return nil, err
-	}
-
-	if !args.Enabled {
-		_, err := repoupdater.DefaultClient.ExcludeRepo(ctx, repo.IDInt32())
```
Comment (tsenart) [inline cmd/frontend/graphqlbackend/repositories.go:294]: We should remove the associated endpoint in repo-updater server too.

Comment (tsenart) [reply cmd/frontend/graphqlbackend/repositories.go:294]: And all the `.Exclude` methods on the `ExternalService` type which I think would be then dead code.

Comment (ryanslade) [reply cmd/frontend/graphqlbackend/repositories.go:294]: You're right that they don't seem to be used anymore, but does adding a repo to the exclude list and disabling a repo mean the same thing? 

We still support adding a repo to the exclude list right?

Comment (tsenart) [reply cmd/frontend/graphqlbackend/repositories.go:294]: > but does adding a repo to the exclude list and disabling a repo mean the same thing?

It's the same thing in practice. When we got rid of the enabled / disabled concept, we realised some customers were still relying on it via the API, so we implemented this programatic "exclude" list manipulation as a replacement, while the customer relied on that.

> We still support adding a repo to the exclude list right?

Yes, we still do that, but AFAIK not programatically anymore after this change, only via external service configuration provided by the user.

Comment (ryanslade) [reply cmd/frontend/graphqlbackend/repositories.go:294]: Cool, thanks for clarifying

**Comment by ryanslade** [comment]
This turned our better than I was expecting 😄 

Screenshot 2021-05-18 at 11 27 51

**Comment by tsenart** [comment]
@ryanslade: Seems like it's gonna be even more :D


```
internal/repos/types_test.go:76:6: func `formatJSON` is unused (unused)
--
&nbsp; | internal/types/types.go:522:26: func `ExternalService.schema` is unused (unused)
&nbsp; | internal/types/types.go:545:27: func `(*ExternalService).validateConfig` is unused (unused)
&nbsp; | internal/types/types.go:496:27: func `(*ExternalService).config` is unused (unused)
```

**Review by tsenart — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ All notable changes to Sourcegraph are documented in this file.
 
 ### Removed
 
--
+- The deprecated `SetRepositoryEnabled` mutation. [#21044](https://github.com/sourcegraph/sourcegraph/pull/21044)
```
Comment (tsenart) [inline CHANGELOG.md:46]: @ryanslade: Please finish this sentence :D

Comment (ryanslade) [reply CHANGELOG.md:46]: I was assuming the "Removed" title would be read first, so it reads:

"Removed the deprecated `SetRepositoryEnabled` mutation."

Comment (ryanslade) [reply CHANGELOG.md]: I updated to make it clearer

**Pull request merged**